src/dr_llm/llm/providers/openai_compat/reasoning.py

```
@@ -2,25 +2,207 @@
+    reasoning_capabilities_for_model,
 )
 
 
-class OpenAICompatReasoningConfig(BaseModel):
-    model_config = ConfigDict(frozen=True)
+def validate_reasoning_for_openai(
+    *, model: str, reasoning: ReasoningSpec | None
+) -> None:
+    def _validate_native(spec: OpenAIReasoning) -> None:
+        if not openai_supports_configurable_thinking(model):
+            raise ValueError(f"openai thinking is not supported for model={model!r}")
```

Ruff (0.15.9)

[warning] Avoid specifying long messages outside the exception class

(TRY003)